### ISSUE TITLE
Brush not filtering when using composite chart

### ISSUE THREAD

**Issue by gramster** [dc-js/dc.js#878]
Is this supposed to work?

If I create a line chart with a data table, selecting an area of the chart with the brush filters the data in the table. If I take that same line chart and wrap it in a composite chart (even if it is the only chart in the composite), the data table no longer gets filtered.

My code is the same in each case except for this:

```
        dcl = dc.lineChart("#dc-chart");
        dcl.dimension(dateDim)
            .group(open, "Open");
```

versus:

```
        dcl = dc.compositeChart("#dc-chart");
        dcl.compose([
                dc.lineChart(dcl)
                    .dimension(dateDim)
                    .group(open, "Open")
      ]);
```

**Comment by gordonwoodhull** [comment]
This is a duplicate of, well, a lot of things. 
https://github.com/dc-js/dc.js/search?q=filter+composite&type=Issues&utf8=%E2%9C%93

This branch is a start on a solution:
https://github.com/dc-js/dc.js/tree/sync-filter

Please try it out and see if it helps.  Will merge to develop soon.

**Comment by jbliesener** [comment]
Any news on this one?

**Comment by Fil** [comment]
I'm stump too

**Comment by Fil** [comment]
This fixed it for me:

https://groups.google.com/forum/#!topic/dc-js-user-group/yI6_cbvgfbU
http://jsfiddle.net/cBgkT/1/

**Comment by joysword** [comment]
As of 2.0.0-beta32, CompositeChart's _chart._brushing() still does not trigger any redraw events. The google group link posted by Fil did not work for me.

A solution that worked for me is to simply comment out CompositeChart's _chart._brushing() function. This makes CompositeCharts use CoordinateGridMixin's _chart._brushing() for brushing, and works just fine.

**Comment by gordonwoodhull** [comment]
Thanks @joysword, that sounds like a good workaround, as long as the dimension is the same for the composite and its children. I think we could make it possible to restore the overridden method without having to change the source.

**Comment by phsstory** [comment]
@Fil  updated for 404'd external resources http://jsfiddle.net/cBgkT/42/

The redefined _chart._brushing function works if the composite graph has the defined dimension but fails if the sub-charts have defined dimensions.

**Comment by cramatt** [comment]
When I override the `_chart._brushing()` method I get unexpected behavior - the filtered set and brush jump around sporadically.  Using 2.0.0-beta32.

**Comment by gordonwoodhull** [comment]
Fixed by #1408 in 3.0.2
